src/core/demangle.d

```diff
@@ -2009,20 +2011,24 @@ pure @safe:
 
 
 /**
- * Demangles D mangled names.  If it is not a D mangled name, it returns its
+ * Demangles D mangled names.  If it is not a recognized mangled name (D, C++), it returns its
  * argument name.
  *
  * Params:
  *  buf = The string to demangle.
  *  dst = An optional destination buffer.
  *
  * Returns:
- *  The demangled name or the original string if the name is not a mangled D
- *  name.
+ *  The demangled name or the original string if the name is not a recognized mangled name.
  */
 char[] demangle( const(char)[] buf, char[] dst = null ) nothrow pure @safe
 {
-    //return Demangle(buf, dst)();
+    // TODO: also report which type: cpp,D,unrecognized
+    // TODO: use same technique as used in demangleCXX for `dst`
+    DemangleCXXStatus status;
```

Comment:
I'd recommend not mixing this into the D demangler. Instead keep it in it's own module, and add a wrapper that forwards to the appropriate demangler.

Reply:
I thought about this but the advantage of doing it in demangle is that all existing tools will automatically benefit from multi-language demangling. I can however add a demangleD (analog to demangleCXX) if needed to only work on D demangling. That would be more symmetrical.